ktor-utils/common/src/io/ktor/util/PlatformUtils.kt

```diff
@@ -4,13 +4,22 @@
 
 package io.ktor.util
 
-public expect object PlatformUtils {
-    public val IS_BROWSER: Boolean
-    public val IS_NODE: Boolean
-    public val IS_JVM: Boolean
-    public val IS_NATIVE: Boolean
+public object PlatformUtils {
+    public val IS_BROWSER: Boolean = platform == Platform.Browser
+    public val IS_NODE: Boolean = platform == Platform.Node
+    public val IS_JVM: Boolean = platform == Platform.Jvm
+    public val IS_NATIVE: Boolean = platform == Platform.Native
```

Comment:
Since these lead to incorrect code, shouldn't they be deprecated?

Also, a more general comment but I don't suggest that it should be changed in this PR: I'm surprised that you can check for `IS_JVM` etc. in common code at all and that in JS code the `Platform` enum also contains `Jvm` which is an impossible case. It feels like `expect`/`actual` are the cleaner solution for those cases and the only real place where you need a runtime check is for Node vs. Browser.

Reply:
We can't drop fields from `enum` based on expect/actual model: we need to refer to all platforms in the common code. There is also a case when we need to check if the platform is native or JVM in the common code, for instance, to use UDP or raw sockets.

Reply:
What I meant was to remove the enum in all other platforms and only have it in JS. And for those cases where you need to use a UDP or raw socket you refer to an `expect fun` which does the platform-specific thing. Is there anything preventing you from just doing it this way?

Reply:
We initially started with the proposed approach. It led to copy-paste expect-actual implementations for many cases, from exception mapping to concurrency. It's hard to maintain.

Making a new platform for different target subsets would be perfect, but it works not so well with tooling.

Reply:
I don't want to deprecate them, because writing `if (PlatformUtils.IS_JVM)` is better then `if (PlatformUtils.platform == Platform.Jvm)` and in the end, we will end up with similar extensions anyway. Also, now `IS_NODE` and `IS_BROWSER` can't be true at the same time, so the problem with flags should be fixed.